### DISCUSSION TITLE
Callback or mechanism to detect the TTL expired

### DISCUSSION THREAD

**Discussion by HectorFire** [lldpd/lldpd#530] [Q&A] [answered]
Hi Vicent,
We are working in a implementation that use your code to discover devices in a network, currently we have a timer to remove a register from a database, but we are looking for a better approach.
Does it exist an API or a way to know when the Time To Live of a neighbor expire?

Thanks for your answer.

Regards
Hector Garrido.

**Comment by vincentbernat** [comment, marked as answer]
You can do that with liblldpctl. See https://github.com/lldpd/lldpd/blob/3020ec35c0a2de36ebe54e9564551c3663be7a3e/src/client/show.c#L192 as an example.
